src/otx/algo/anomaly/openvino_model.py

```diff
@@ -133,6 +137,55 @@ def _create_model(self) -> Model:
             configuration=self.model_api_configuration,
         )
 
+    def optimize(
+        self,
+        output_dir: Path,
+        data_module: OTXDataModule,
+        ptq_config: dict[str, Any] | None = None,
+    ) -> Path:
+        """Runs NNCF quantization.
+
+        Note:
+            The only difference between the base class is that it uses `val_dataloader` instead of `train_dataloader`.
+
+        See ``otx.core.model.base.OVModel.optimize`` for more details.
+        """
+        import nncf
+
+        output_model_path = output_dir / (self._OPTIMIZED_MODEL_BASE_NAME + ".xml")
+
+        def check_if_quantized(model: openvino.Model) -> bool:
+            """Checks if OpenVINO model is already quantized."""
+            nodes = model.get_ops()
+            return any(op.get_type_name() == "FakeQuantize" for op in nodes)
+
+        ov_model = openvino.Core().read_model(self.model_name)
+
+        if check_if_quantized(ov_model):
+            msg = "Model is already optimized by PTQ"
+            raise RuntimeError(msg)
+
+        train_dataset = data_module.val_dataloader()
```

Comment:
Since this is the only line that is different is there a better way to update this method without duplicating the code?

Reply:
Is there any other reason why Anomaly specifically uses `val_dataloader`?
@sovrasov What do you think about this? If this is what we need, shouldn't we be able to specify a subset in `OVModel.optimize`?

Reply:
I am not the expert here. I just referred to Adrian's code here https://github.com/openvinotoolkit/anomalib/blob/e5b91d6a32c469ec0b9c4c161272b68114ba6fa3/src/anomalib/models/components/base/export_mixin.py#L245
The only difference is that the val dataset also contains the anomaly class.